modules/objdetect/include/opencv2/objdetect/aruco_board.hpp

```diff
@@ -0,0 +1,310 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+#ifndef OPENCV_ARUCO_BOARD_HPP
+#define OPENCV_ARUCO_BOARD_HPP
+
+#include <opencv2/core.hpp>
+
+namespace cv {
+namespace aruco {
+//! @addtogroup aruco
+//! @{
+
+class Dictionary;
+
+/**
+ * @brief Board of markers
+ *
+ * A board is a set of markers in the 3D space with a common coordinate system.
+ * The common form of a board of marker is a planar (2D) board, however any 3D layout can be used.
+ * A Board object is composed by:
+ * - The object points of the marker corners, i.e. their coordinates respect to the board system.
+ * - The dictionary which indicates the type of markers of the board
+ * - The identifier of all the markers in the board.
+ */
+class CV_EXPORTS_W Board {
+public:
+    CV_WRAP Board();
+
+    /**
+     * @brief Draw a planar board
+     *
+     * @param outSize size of the output image in pixels.
+     * @param img output image with the board. The size of this image will be outSize
+     * and the board will be on the center, keeping the board proportions.
+     * @param marginSize minimum margins (in pixels) of the board in the output image
+     * @param borderBits width of the marker borders.
+     *
+     * This function return the image of the GridBoard, ready to be printed.
+     */
+    CV_WRAP virtual void draw(Size outSize, OutputArray img, int marginSize = 0, int borderBits = 1) const;
```

Comment:
AFAIK, there is some exception in parameters order for "drawing" functions.

/cc @vpisarev

Reply:
I take a look on implementation. It is not a classical "drawing" function, so such name is confusing.

Proper naming for such functionality is `generate` / `generateImage` (like https://github.com/opencv/opencv/blob/4.6.0/doc/pattern_tools/gen_pattern.py )

Reply:
renaming `draw()` to `generateImage()`
renaming `drawMarker()` to `generateMarkerImage()`